From 9ed3dc901c485b29de9cc11d473bfda9da3c6561 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:38 -0400
Subject: [tools] perf/kmem: Introduce --list-cmds for use by scripts

Message-id: <1410181752-25631-154-git-send-email-jolsa@redhat.com>
Patchwork-id: 92000
O-Subject: [PATCH RHEL7.1 BZ1134356 153/307] perf kmem: Introduce --list-cmds for use by scripts
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 3bca23543b910a84d3c58a819663e0a4fad72bb6
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Fri Mar 14 23:17:51 2014 -0400
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-kmem.c b/tools/perf/builtin-kmem.c
index 929462a..bd91de0 100644
--- a/tools/perf/builtin-kmem.c
+++ b/tools/perf/builtin-kmem.c
@@ -756,11 +756,13 @@ int cmd_kmem(int argc, const char **argv, const char *prefix __maybe_unused)
  OPT_BOOLEAN(0, "raw-ip", &raw_ip, "show raw ip instead of symbol"),
  OPT_END()
  };
- const char * const kmem_usage[] = {
-  "perf kmem [<options>] {record|stat}",
+ const char *const kmem_subcommands[] = { "record", "stat", NULL };
+ const char *kmem_usage[] = {
+  NULL,
   NULL
  };
- argc = parse_options(argc, argv, kmem_options, kmem_usage, 0);
+ argc = parse_options_subcommand(argc, argv, kmem_options,
+     kmem_subcommands, kmem_usage, 0);
 
  if (!argc)
   usage_with_options(kmem_usage, kmem_options);
diff --git a/tools/perf/perf-completion.sh b/tools/perf/perf-completion.sh
index ae3a576..0ef59dd 100644
--- a/tools/perf/perf-completion.sh
+++ b/tools/perf/perf-completion.sh
@@ -121,8 +121,8 @@ __perf_main ()
  elif [[ $prev == "-e" && "${words[1]}" == @(record|stat|top) ]]; then
   evts=$($cmd list --raw-dump)
   __perfcomp_colon "$evts" "$cur"
- # List subcommands for 'perf kvm'
- elif [[ $prev == "kvm" ]]; then
+ # List subcommands for perf commands
+ elif [[ $prev == @(kvm|kmem) ]]; then
   subcmds=$($cmd $prev --list-cmds)
   __perfcomp_colon "$subcmds" "$cur"
  # List long option names
-- 
1.7.1